### DISCUSSION TITLE
Building on MacOS fails for 6.3.1

### DISCUSSION THREAD

**Discussion by romanbsd** [puma/puma#3217] [Q&A]
**Describe the bug**
    current directory: /Users/roman/.rvm/gems/ruby-3.0.6/gems/puma-6.3.1/ext/puma_http11
/Users/roman/.rvm/rubies/ruby-3.0.6/bin/ruby -I /Users/roman/.rvm/rubies/ruby-3.0.6/lib/ruby/3.0.0 -r ./siteconf20230830-35341-aolk9f.rb extconf.rb
using OpenSSL pkgconfig (openssl.pc)
checking for openssl/bio.h... yes
checking for DTLS_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_session_cache_mode(NULL, 0) in openssl/ssl.h... yes
checking for TLS_server_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_min_proto_version(NULL, 0) in openssl/ssl.h... yes
checking for X509_STORE_up_ref()... yes
checking for SSL_CTX_set_ecdh_auto(NULL, 0) in openssl/ssl.h... yes
checking for SSL_CTX_set_dh_auto(NULL, 0) in openssl/ssl.h... yes
checking for SSL_get1_peer_certificate() in openssl/ssl.h... no
checking for Random.bytes... yes
creating Makefile

current directory: /Users/roman/.rvm/gems/ruby-3.0.6/gems/puma-6.3.1/ext/puma_http11
make DESTDIR\= clean

current directory: /Users/roman/.rvm/gems/ruby-3.0.6/gems/puma-6.3.1/ext/puma_http11
make DESTDIR\=
compiling http11_parser.c
compiling mini_ssl.c
compiling puma_http11.c
linking shared-object puma/puma_http11.bundle
Undefined symbols for architecture arm64:
  "_SSL_get1_peer_certificate", referenced from:
      _engine_peercert in mini_ssl.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [puma_http11.bundle] Error 1

**Desktop (please complete the following information):**
 - OS: Mac
 - Puma Version 6.3.1

**Comment by dentarg** [comment]
Please search existing issues and discussions before opening a new issue.

See https://github.com/puma/puma/issues/2790#issuecomment-1547332463

**Reply by cesc1989** [reply]
Thanks for linking
